Formula/s2geometry.rb

```diff
@@ -0,0 +1,91 @@
+class S2geometry < Formula
+  desc "Computational geometry and spatial indexing on the sphere"
+  homepage "https://github.com/google/s2geometry.git"
+  url "https://github.com/google/s2geometry/archive/v0.9.0.tar.gz"
+  sha256 "54c09b653f68929e8929bffa60ea568e26f3b4a51e1b1734f5c3c037f1d89062"
+
+  depends_on "cmake" => :build
+  depends_on "glog" => :build
+  depends_on "ninja" => :build
+  depends_on "openssl"
+
+  resource "gtest" do
+    url "https://github.com/google/googletest/archive/release-1.8.1.tar.gz"
+    sha256 "9bf1fe5182a604b4135edc1a425ae356c9ad15e9b23f9f12a02e80184c3a249c"
+  end
+
+  def install
+    ENV["OPENSSL_ROOT_DIR"] = Formula["openssl"].opt_prefix
+
+    (buildpath/"gtest").install resource "gtest"
+    (buildpath/"gtest/googletest").cd do
+      system "cmake", "."
+      system "make"
+    end
+    ENV["CXXFLAGS"] = "-I../gtest/googletest/include"
+
+    mkdir "build" do
+      args = std_cmake_args
+      args << "-DWITH_GLOG=1"
+      args << "-DBUILD_SHARED_LIBS=OFF"
+      args << "-DOPENSSL_USE_STATIC_LIBS=TRUE"
+      args << "-DCMAKE_BUILD_TYPE=Release"
+      args << "-DCMAKE_OSX_SYSROOT=/" unless MacOS::Xcode.installed?
+      args << ".."
+      system "cmake", "-G", "Unix Makefiles", *args
+      system "make", "install"
+    end
+    mkdir "build-shared" do
+      args = std_cmake_args
+      args << "-DCMAKE_BUILD_TYPE=Release"
+      args << "-DWITH_GLOG=1"
+      args << "-DCMAKE_OSX_SYSROOT=/" unless MacOS::Xcode.installed?
```

Comment:
Flags common to shared and static cmake invocations could be put above, in an `args` variable.

Reply:
shared flags have been factored out